contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1301 @@
+            "Not the Governor or Strategist"
+        );
+        _;
+    }
+
+    /// @notice the constructor
+    /// @dev This contract is intended to be used as a proxy. To prevent the
+    ///      potential confusion of having a functional implementation contract
+    ///      the constructor has the `initializer` modifier. This way the
+    ///      `initialize` function can not be called on the implementation contract.
+    ///      For the same reason the implementation contract also has the governor
+    ///      set to a zero address.
+    /// @param _stratConfig the basic strategy configuration
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _oethpAddress Address of the Erc20 OETHp Token contract
+    /// @param _liquidityManager Address of liquidity manager to add
+    ///         the liquidity
+    /// @param _poolLens Address of the pool lens contract
+    /// @param _maverickPosition Address of the Maverick's position contract
+    /// @param _maverickQuoter Address of the Maverick's Quoter contract
+    /// @param _mPool Address of the Aerodrome concentrated liquidity pool
+    /// @param _upperTickAtParity Bool when true upperTick is the one where the
+    ///        price of OETHp and WETH are at parity
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _wethAddress,
+        address _oethpAddress,
+        address _liquidityManager,
+        address _poolLens,
+        address _maverickPosition,
+        address _maverickQuoter,
+        address _mPool,
+        bool _upperTickAtParity,
+        address _votingDistributor,
+        address _poolDistributor
+    ) initializer InitializableAbstractStrategy(_stratConfig) {
+        require(
+            address(IMaverickV2Pool(_mPool).tokenA()) == _wethAddress,
+            "Only WETH supported as tokenA"
+        );
+        require(
+            address(IMaverickV2Pool(_mPool).tokenB()) == _oethpAddress,
+            "Only OETHp supported as tokenB"
+        );
+        require(
+            _liquidityManager != address(0),
+            "LiquidityManager zero address not allowed"
+        );
+        require(
+            _maverickQuoter != address(0),
+            "Quoter zero address not allowed"
+        );
+        require(_poolLens != address(0), "PoolLens zero address not allowed");
+        require(
+            _maverickPosition != address(0),
+            "Position zero address not allowed"
+        );
+        require(
+            _votingDistributor != address(0),
+            "Voting distributor zero address not allowed"
+        );
+        require(
+            _poolDistributor != address(0),
+            "Pool distributor zero address not allowed"
+        );
+
+        uint256 _tickSpacing = IMaverickV2Pool(_mPool).tickSpacing();
+        require(_tickSpacing == 1, "Unsupported tickSpacing");
+        // -1 is the tick where the contract deposits liquidity to
+        tickNumber = -1;
```

Comment:
Just by curiosity, why not use a constant here instead of immutable? For the sake of harmony?

Reply:
Agreed constant would be more correct: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/7a970ed694b618b77937cd63283e1ef2d23ec860